Duplicate template instantiation prevents building on armv7 (Raspberry Pi 4)

# Issue

Previously I successfully built transmission from 5026d4392 on my Raspberry Pi 4.
Trying to build from d31254df5 today, I am confronted with the following error.

```
/home/karl/AUR/transmission-cli-git/src/transmission/libtransmission/utils.cc:1423:74: error: duplicate explicit instantiation of ‘std::optional<_Tp> tr_parseNum(std::string_view&, int) [with T = unsigned int; typename std::enable_if<std::is_integral<_Tp>::value, bool>::type <anonymous> = true; std::string_view = std::basic_string_view<char>]’ [-fpermissive]
```
After simply deleting the offending lines in `utils.cc`, https://github.com/transmission/transmission/blob/2293f4336af1abe52a487fe1d4c6b55ce35c007d/libtransmission/utils.cc#L1421-L1423
the build succeeds.

I also get many warnings about a redundant redeclaration,
```
In file included from /home/karl/AUR/transmission-cli-git/src/transmission/libtransmission/torrent-files.h:20,
                 from /home/karl/AUR/transmission-cli-git/src/transmission/libtransmission/torrent-metainfo.h:18,
                 from /home/karl/AUR/transmission-cli-git/src/transmission/libtransmission/announce-list.cc:15:
/home/karl/AUR/transmission-cli-git/src/transmission/libtransmission/file.h:257:18: warning: redundant redeclaration of ‘std::string_view tr_sys_path_dirname(std::string_view)’ in same scope [-Wredundant-decls]
  257 | std::string_view tr_sys_path_dirname(std::string_view path);
      |                  ^~~~~~~~~~~~~~~~~~~
In file included from /home/karl/AUR/transmission-cli-git/src/transmission/libtransmission/web-utils.h:15,
                 from /home/karl/AUR/transmission-cli-git/src/transmission/libtransmission/announce-list.h:18,
                 from /home/karl/AUR/transmission-cli-git/src/transmission/libtransmission/announce-list.cc:13:
/home/karl/AUR/transmission-cli-git/src/transmission/libtransmission/tr-strbuf.h:13:18: note: previous declaration of ‘std::string_view tr_sys_path_dirname(std::string_view)’
   13 | std::string_view tr_sys_path_dirname(std::string_view path);
      |                  ^~~~~~~~~~~~~~~~~~~
```

Neither issue is present building on my x86_64 system.

# Version info

```
% uname -a
Linux schwarzschild 5.15.24-1-rpi-ARCH #1 SMP Fri Feb 18 16:18:00 UTC 2022 armv7l GNU/Linux
% gcc --version
gcc (GCC) 11.2.0
Copyright (C) 2021 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
% head /proc/cpuinfo
processor       : 0
model name      : ARMv7 Processor rev 3 (v7l)
BogoMIPS        : 270.00
Features        : half thumb fastmult vfp edsp neon vfpv3 tls vfpv4 idiva idivt vfpd32 lpae evtstrm crc32
CPU implementer : 0x41
CPU architecture: 7
CPU variant     : 0x0
CPU part        : 0xd08
CPU revision    : 3
% tail -n 4 /proc/cpuinfo
Hardware        : BCM2711
Revision        : c03111
Serial          : 100000007d3f9dc1
Model           : Raspberry Pi 4 Model B Rev 1.1
```

--- a/libtransmission/session.cc
+++ b/libtransmission/session.cc
@@ -798,7 +798,7 @@ static void sessionSetImpl(struct init_data* const data)
     if (tr_variantDictFindStrView(settings, TR_KEY_umask, &sv))
     {
         /* Read a umask as a string representing an octal number. */
-        session->umask = tr_parseNum<mode_t>(sv, 8).value_or(DefaultUmask);
+        session->umask = static_cast<mode_t>(tr_parseNum<uint32_t>(sv, 8).value_or(DefaultUmask));
         umask(session->umask);
     }
     else if (tr_variantDictFindInt(settings, TR_KEY_umask, &i))

--- a/libtransmission/utils.cc
+++ b/libtransmission/utils.cc
@@ -1416,12 +1416,12 @@ template<typename T, std::enable_if_t<std::is_integral<T>::value, bool>>
 #endif // #if defined(__GNUC__) && !__has_include(<charconv>)
 
 template std::optional<int64_t> tr_parseNum(std::string_view& sv, int base);
-template std::optional<int> tr_parseNum(std::string_view& sv, int base);
-template std::optional<size_t> tr_parseNum(std::string_view& sv, int base);
+template std::optional<int32_t> tr_parseNum(std::string_view& sv, int base);
+template std::optional<int8_t> tr_parseNum(std::string_view& sv, int base);
 
-#ifndef _WIN32
-template std::optional<mode_t> tr_parseNum(std::string_view& sv, int base);
-#endif
+template std::optional<uint64_t> tr_parseNum(std::string_view& sv, int base);
+template std::optional<uint32_t> tr_parseNum(std::string_view& sv, int base);
+template std::optional<uint8_t> tr_parseNum(std::string_view& sv, int base);
 
 template<typename T, std::enable_if_t<std::is_floating_point<T>::value, bool>>
 [[nodiscard]] std::optional<T> tr_parseNum(std::string_view& sv)

